client/homebrew/pages/basePages/listPage/listPage.jsx

```diff
@@ -74,19 +75,39 @@ const ListPage = createClass({
 
 	handleFilterTextChange : function(e){
 		this.setState({
-			filterString : e.target.value
+			filterString : e.target.value,
 		});
 		return;
 	},
 
+	handleKeys : function(e){
+		if(e.key === 'Enter') {
+			this.updateUrl(e.target.value);
+		  }
+	},
+
+	updateUrl : function(filterTerm){
+		const url = new URL(window.location.href);
+		const urlParams = new URLSearchParams(url.search);
+		if(urlParams.get('filter') == filterTerm) return;
+		urlParams.set('filter', filterTerm);
+		if(!filterTerm) urlParams.delete('filter');
+		url.search = urlParams;
+		window.location.replace(url.href);
```

Comment:
IIRC, updating `window.location` will cause a page reload (which might reset other widgets on the page, such as sort order).
 
What you might want here is instead to use `history.replaceState()`.
https://developer.mozilla.org/en-US/docs/Web/API/History/replaceState

Reply:
In that case, I can probably remove the 'onKeyDown ()=> if(keyDown=Enter) updateUrl()` stuff and just do it live with `history.replaceState()`... Hang on, I'm just going to quickly rewrite half of this PR...

Reply:
I vastly prefer the performance of `history.replaceState` to `window.location.replace`.

The URL now updates live.

Reply:
A small thing, but I like that `history.replaceState` permits the use of a URL object, where `window.location.replace` needed to be passed `url.href` specifically.